Userland/Libraries/LibGL/Tex/Sampler2D.cpp

```diff
@@ -21,12 +21,12 @@ static constexpr float wrap_mirrored_repeat(float value)
     float integer = floorf(value);
     float frac = value - integer;
     bool iseven = fmodf(integer, 2.0f) == 0.0f;
-    return iseven ? frac : 1 - frac;
+    return iseven ? frac : 1 - frac - NumericLimits<float>::epsilon();
 }
 
 static constexpr float wrap_clamp(float value)
 {
-    return clamp(value, 0.0f, 1.0f);
+    return clamp(value, 0.0f, 1.0f - NumericLimits<float>::epsilon());
```

Comment:
On page 272 of [the spec](https://www.khronos.org/registry/OpenGL/specs/gl/glspec15.pdf) it is mentioned that the range is exactly `[0, 1]` and the texture's border should be taken into account when sampling - I don't think that is implemented yet.

Reply:
Maybe you could create a special case for `1.0` texture coordinates after wrapping and add a `FIXME`?

Reply:
We use `GL_CLAMP_TO_EDGE` behaviour for now.